src/types/ErrorRate/__tests__/GraphEdgeStatus.test.ts

```diff
@@ -0,0 +1,141 @@
+import { setServerConfig } from '../../../config/ServerConfig';
+import { serverRateConfig } from '../__testData__/ErrorRateConfig';
+import { calculateStatusGraph, generateRateForGraphTolerance, getTotalRequest } from '../GraphEdgeStatus';
+import { Responses } from '../../Graph';
+import { RequestTolerance } from '../types';
+import { ToleranceConfig } from '../../ServerConfig';
+import * as H from '../../Health';
+
+const tolerance400: ToleranceConfig = {
+  code: new RegExp(/400/),
+  degraded: 10,
+  failure: 20,
+  protocol: new RegExp(/http/),
+  direction: new RegExp(/inbound/)
+};
+
+const tolerance500: ToleranceConfig = {
+  code: new RegExp(/500/),
+  degraded: 10,
+  failure: 20,
+  protocol: new RegExp(/http/),
+  direction: new RegExp(/inbound/)
+};
+
+describe('getRateHealthConfig', () => {
+  beforeAll(() => {
+    setServerConfig(serverRateConfig);
+  });
+
+  it('getTotalRequest', () => {
+    // Check with annotation to set 400
+    const resp: Responses = {
+      '200': { hosts: {}, flags: { '-': '3', XX: '4', YYY: '2' } },
+      '400': { hosts: {}, flags: { '-': '1', XX: '1', YYY: '6' } }
+    };
+    expect(getTotalRequest(resp)).toBe(17);
+  });
+  describe('calculateStatusGraph', () => {
```

Comment:
These tests are good.
Perhaps we'll add more in the future as I can find some cases that worth to add.
But good to have them, otherwise it's hard to review the GraphEdgeStatus and validate if there is anything missing.